test/e2e/scheduling/preemption.go

```diff
@@ -92,13 +92,29 @@ var _ = SIGDescribe("SchedulerPreemption [Serial]", func() {
 		// Create one pod per node that uses a lot of the node's resources.
 		By("Create pods that use 60% of node resources.")
 		pods := make([]*corev1.Pod, len(nodeList.Items))
+		allPods, err := cs.CoreV1().Pods(metav1.NamespaceAll).List(metav1.ListOptions{})
 		for i, node := range nodeList.Items {
+			// There is a chance that node.Status.Allocatable has changed between the time we created node objects
+			// and now, so better to calculate the current usage again before assigning. This can happen when pods
+			// frequently jump from one node to another either because of eviction or preemption. So better to get
+			// current value
+			currentCpuUsage, currentMemUsage := getCurrentPodUsageOnTheNode(node.Name, allPods.Items, podRequestedResource)
 			cpuAllocatable, found := node.Status.Allocatable["cpu"]
 			Expect(found).To(Equal(true))
-			milliCPU := cpuAllocatable.MilliValue() * 40 / 100
+			milliCPU := cpuAllocatable.MilliValue()
+			if milliCPU != currentCpuUsage {
+				framework.Logf("Allocatable is not equal to current usage change it to current value for node %v", node.Name)
+				milliCPU = currentCpuUsage
+			}
+			milliCPU = milliCPU * 40 / 100
 			memAllocatable, found := node.Status.Allocatable["memory"]
 			Expect(found).To(Equal(true))
-			memory := memAllocatable.Value() * 60 / 100
+			memory := memAllocatable.Value()
+			if memory != currentMemUsage {
+				framework.Logf("Allocatable is not equal to current usage change it to current value for node %v", node.Name)
+				memory = currentMemUsage
+			}
+			memory = memory * 60 / 100
```

Comment:
btw @bsalamat - after looking at the tests I am wondering if we should allow direct assignment of 3 pods to nodes instead of letting scheduler schedule them, we can let scheduler preempt one of them for required pod at https://github.com/kubernetes/kubernetes/pull/76663/files#diff-e35316a06970f9000b7b154addabda8cL125

Reply:
We could, but there is no real need to change it.